services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/api/compose.py

```diff
@@ -0,0 +1,167 @@
+import logging
+import traceback
+from typing import Optional
+
+from fastapi import APIRouter, Request, Response
+from fastapi.responses import PlainTextResponse
+
+from ..settings import ServiceSidecarSettings
+from ..shared_handlers import remove_the_compose_spec, write_file_and_run_command
+from ..storage import SharedStore
+from ..utils import InvalidComposeSpec
+
+logger = logging.getLogger(__name__)
+compose_router = APIRouter()
+
+
+@compose_router.post(
+    "/compose:store", response_class=PlainTextResponse, responses={204: {"model": None}}
+)
+async def store_docker_compose_spec_for_later_usage(
+    request: Request, response: Response
+) -> Optional[str]:
+    """ Expects the docker-compose spec as raw-body utf-8 encoded text """
+    body_as_text = (await request.body()).decode("utf-8")
+
+    shared_store: SharedStore = request.app.state.shared_store
+
+    try:
+        shared_store.put_spec(body_as_text)
+    except InvalidComposeSpec as e:
+        logger.warning("Error detected %s", traceback.format_exc())
+        response.status_code = 400
+        return str(e)
+
+    response.status_code = 204
+    return None
+
+
+@compose_router.post("/compose:preload", response_class=PlainTextResponse)
+async def create_docker_compose_configuration_containers_without_starting(
+    request: Request, response: Response, command_timeout: float
+) -> str:
+    """ Expects the docker-compose spec as raw-body utf-8 encoded text """
+    body_as_text = (await request.body()).decode("utf-8")
+
+    settings: ServiceSidecarSettings = request.app.state.settings
+    shared_store: SharedStore = request.app.state.shared_store
+
+    try:
+        shared_store.put_spec(body_as_text)
+    except InvalidComposeSpec as e:
+        logger.warning("Error detected %s", traceback.format_exc())
+        response.status_code = 400
+        return str(e)
+
+    # --no-build might be a security risk building is disabled
+    command = "docker-compose -p {project} -f {file_path} up --no-build --no-start"
```

Comment:
please use verbose arguments. this is similar as in a script. see [this](https://kvz.io/bash-best-practices.html)